### PR TITLE
Allow getDBSaltForKey error to propagate for fxiOS

### PR DESCRIPTION
**Pull request by skhamis**
The way iOS uses getDBSaltForKey, they have a do/catch wrapper if there is no salt and generate a new one [Here](https://github.com/mozilla-mobile/firefox-ios/blob/041c85483b1aa4a4e357d8bd9419550e4f20347a/Storage/Rust/RustLogins.swift#L108). The uniffi logins effort added a `try!` which hard fails and causes error propagation to be disabled.

I have also updated the iOS testing on the a-s side to better mimic what fxiOS is doing and testy the APIs the way they are using it.

### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

### CODE REVIEW

File: megazords/ios/MozillaAppServicesTests/LoginsTests.swift [hunk 1]
```diff
@@ -6,36 +6,60 @@ import XCTest
 @testable import MozillaAppServices
 
 class LoginsTests: XCTestCase {
-    func getTestStorage() -> LoginsStorage {
+    var storage: LoginsStorage!
+    var encryptionKey: String!
+    var salt: String!
+
+    //This tests setup mimics closer to what fxiOS is doing
```
Comment (rfk) [inline megazords/ios/MozillaAppServicesTests/LoginsTests.swift:13]: nit: "closer" is good context for reviewing the PR, but in a few months time someone reading this is just going to think "closer than what?"; something like "as closely as we can" or similar might be better for a long-lived comment.

**Review by rfk — APPROVED** [review]
Makes sense to me 👍🏻

**Comment by rfk** [comment]
@skhamis I was about to cut a release for unrelated reasons, do you want to get this fix into a new release as well? (ref https://github.com/mozilla/application-services/pull/4206)

**Comment by skhamis** [comment]
> @skhamis I was about to cut a release for unrelated reasons, do you want to get this fix into a new release as well? (ref #4206)

@rfk Actually that'd be awesome! let me make that quick nit and swiftlint error and get this merged in ASAP

**Pull request merged**